tests/tests/Data/RawCodeCoverageDataTest.php

```diff
@@ -333,7 +331,6 @@ public function testHeavyIndentationIsHandledCorrectly(): void
 
         $this->assertEquals(
             [
-                9,
```

Comment:
Variable assignment must be counted as executable line

Reply:
the whole assign: https://github.com/sebastianbergmann/php-code-coverage/blob/384d88deeba1a1b9635b045ed244c8c6b866bab7/tests/_files/source_with_heavy_indentation.php#L9-L13

produces only 1 executable/coverage line in xdebug - line 12

line 9 is not present

Reply:
In my checks it does:
```php
<?php

class Foo
{
    private $state = 1;

    public function isOne(): void
    {
        $v3
            =
            (bool)
            AType::OPT
        ;
    }
}

class AType
{
    const OPT = 1;
}
```
xDebug report:
```
    [./tests/_files/source_with_heavy_indentation.php] => Array
        (
            [9] => 1
            [12] => 1
            [14] => 1
            [20] => 1
        )
```
As it always did, and mandatory to have an exec line for such conditions.

Reply:
What is your xdebug/PHP version? Isn't this the exact problem https://github.com/sebastianbergmann/php-code-coverage/issues/889 why you added executable lines, ie. what is your output /wo `AType` defined (and `isOne` never called - this is how I tested it)?

https://github.com/sebastianbergmann/php-code-coverage/pull/949#issuecomment-1301744174 - do you know the `Compare` button after force-push in GH? it nothing has changed, the compare result is empty.

Reply:
```php
<?php

class Foo
{
    private $state = 1;

    public function isOne(): void
    {
        $v3
            =
            (bool)
            1
        ;
    }
}

(new Foo())->isOne();
```
```console
$ XDEBUG_MODE=coverage php -r 'xdebug_start_code_coverage();$file = "tests/_files/source_with_heavy_indentation.php";require $file;print_r(xdebug_get_code_coverage()[$file]);'
Array
(
    [9] => 1
    [12] => 1
    [14] => 1
    [17] => 1
)

$ php -v
PHP 8.1.11 (cli) (built: Sep 29 2022 22:29:14) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.11, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.11, Copyright (c), by Zend Technologies
    with Xdebug v3.1.5, Copyright (c) 2002-2022, by Derick Rethans
```

Reply:
in
```
PHP 7.4.19 (cli) (built: May  4 2021 14:24:16) ( NTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.19, Copyright (c), by Zend Technologies
    with Xdebug v3.0.4, Copyright (c) 2002-2021, by Derick Rethans
```

and script:
```
<?php

$file = realpath(__DIR__ . '/../../phpunit/php-code-coverage/tests/_files/source_with_heavy_indentation.php');

xdebug_start_code_coverage(XDEBUG_CC_UNUSED | XDEBUG_CC_DEAD_CODE);
require $file;
print_r(xdebug_get_code_coverage()[$file]);
```

I get:

```
Array
(
    [12] => -1
    [16] => -1
    [18] => -1
    [19] => -1
    [20] => -1
    [21] => -1
    [22] => -1
    [24] => -1
...
```

Reply:
Different versions of PHP compile sourcecode to different bytecode. Different bytecode means different code coverage results from Xdebug or PCOV.

Reply:
Indeed, line `9` is considered executable line only from PHP 8.0, but in PHP 7.4 it isn't.

What would you like to do about this @sebastianbergmann ?
Different `php-code-coverage` behaviour for different PHP versions, of just push for the most recent version?

Reply:
I think we should use the behaviour of the current stable PHP version.

Reply:
@Slamdunk any idea how line 9 can be added? I tried adding `PhpParser\Node\Expr\Variable` node as executable as proposed in https://github.com/sebastianbergmann/php-code-coverage/pull/955 but then the "computeReturns" is broken and line 12 is not reported

see my last commit with a test case, assign can be more complex due array desctructing and both lines 97 and 98 are present in xdebug (tested with PHP 8.1)

Reply:
```
        $v2
            =
            (bool)
            1
        ;
```
is very interesting test case

line 1 or line 4 is reported, depending if cached (in opcached) or not, so assign is not stable enought to produce any executable line - it makes sense, no user code is executed

Reply:
@Slamdunk PR is done, before I squash the changes, please review the changes

Reply:
documented in tests